DragonFruit.Six.API/Data/SeasonStats.cs

```diff
@@ -77,5 +78,16 @@ public class SeasonStats
 
         [JsonProperty("last_match_skill_stdev_change")]
         public double LastMatchSkillUncertaintyChange { get; set; }
+
+        [JsonProperty("rank_info")]
+        public RankContainer RankInfo
+        {
+            get
+            {
+                return Rank > 16 ? References.Ranks((int)Rank) : References.LegacyRanks((int)Rank);
+            }
+
+            set => value ??= Rank > 16 ? References.Ranks((int)Rank) : References.LegacyRanks((int)Rank);
```

Comment:
there's no need for get and set. It needs a backing field like:

```
private RankContainer _rankInfo;

public RankContainer RankInfo => _rankInfo ??= Rank > 16 ? References.Ranks((int)Rank) : References.LegacyRanks((int)Rank);
```

where the `private` backing field is at the top of the class (above all)

Reply:
Interesting. I am normally accustomed to only using a public property with getters and setters but this looks great. Thank you for showing this new type of assignment!